### ISSUE TITLE
Pedometer 2.1.0

### ISSUE THREAD

**Issue by sumeetkumar1** [cph-cachet/flutter-plugins#329]
**Remember to specify the plugin name in the title!**

### Device
 - Device: Oneplus 7 Pro
 - OS: Android version 11 Build number Oxygen Os 11.GM21AA

### Describe the bug
I have run this same app on a rooted Samsung device the function seems to be working fine but on a non rooted oneplus 7 pro this is the error I am getting: 
E/SensorManager(22656): registerListenerImpl sensorName:pedometer  Non-wakeup,isWakeUpSensor:false,callingApp: application_name ,callingPid:22656,callingUid:10526
E/SensorManager(22656): registerListenerImpl sensorName:pedometer  Non-wakeup,isWakeUpSensor:false,callingApp: application_name,callingPid:22656,callingUid:10526

### Expected behavior
to show the step count. 

I am not able to understand this error I am not sure what to do here.

**Comment by sumeetkumar1** [comment]
I have found a fix for this issue. The User needs to grant permission to the application at run time in order for the application to read sensor data.

**Comment by LorenzoMonti** [comment]
Hi @sumeetkumar1, I granted permission from the settings. I also used the permission handler package (https://pub.dev/packages/permission_handler) to prove it, but the SensorManager error remains.

**DEVICE**
- Device: OP 6
- OS: Android version 10, Oxygen Os 10.3.9

Any ideas?

**Comment by sumeetkumar1** [comment]
Hey, @LorenzoMonti ok I've used the same permission handler as well. if you can share the error log I could have a Look.

**Comment by LorenzoMonti** [comment]
For sure!

This is the error log in output:

Launching lib/main.dart on ONEPLUS A6000 in debug mode...
Running Gradle task 'assembleDebug'...
✓ Built build/app/outputs/flutter-apk/app-debug.apk.
Debug service listening on was:...
Syncing files to device ONEPLUS...
I/flutter (32766): Permission activity Recognition: granted **[this is the output from permission_handler]**
E/SensorManager(32766): registerListenerImpl sensorName:pedometer  Non-wakeup,isWakeUpSensor:false,callingApp: com.example.pedometer,callingPid:32766,callingUid:10390
E/SensorManager(32766): registerListenerImpl sensorName:pedometer  Non-wakeup,isWakeUpSensor:false,callingApp: com.example.pedometer,callingPid:32766,callingUid:10390
I/flutter (32766): Steps taken: 25015 at 2021-04-22T15:12:28.728034

**Comment by LorenzoMonti** [comment]
Ok, I did some tests and it seems to be just a "latency problem". It works (and it's pretty accurate ) but only after ~20 steps.

**Comment by sumeetkumar1** [comment]
Yup the sensor in the Oneplus phone sends the data in batches it will take some time to show up.  Though This is not true for all phones, in one of the Samsung phones I tested the sensor spits out data almost instantly.

**Comment by LorenzoMonti** [comment]
Cool, thank you.
